packages/amplify-graphql-model-transformer/rds-lambda/handler.ts

```diff
@@ -1,6 +1,49 @@
-import { mySqlPasswordClient } from './clients/mysql/MySQLPasswordClient';
+import { SSMClient, GetParameterCommand } from '@aws-sdk/client-ssm';
+// @ts-ignore
+import { DBAdapter, DBConfig, getDBAdapter } from 'rds-query-processor';
 
-export const run = async (event) => {
-  const result = await mySqlPasswordClient.executeRequest(event);
+let adapter: DBAdapter;
+let secretsClient: SSMClient;
+
+export const run = async (event): Promise<any> => {
+  if (!adapter) {
+    const config = await getDBConfig();
+    adapter = await getDBAdapter(config);
+  }
+  const result = await adapter.executeRequest(event);
   return result;
 };
+
+const createSSMClient = (): void => {
+  secretsClient = new SSMClient({});
+}
+
+const getSSMValue = async(key: string | undefined): Promise<string> => {
+  if (!key) {
+    throw Error('Key not provided to retrieve database connection secret');
+  }
+  const parameterCommand = new GetParameterCommand({
+    Name: key,
+    WithDecryption: true,
+  });
+  const data = await secretsClient.send(parameterCommand);
+  if ((data.$metadata?.httpStatusCode && data?.$metadata?.httpStatusCode >= 400) || !data.Parameter?.Value) {
+    throw new Error('Unable to get secret for database connection');
+  }
+  return data.Parameter?.Value ?? '';
+}
+
+const getDBConfig = async(): DBConfig  => {
+  if (!secretsClient) {
```

Comment:
This method is doing 2 things: creating SSM client and getting DB config. Moving the SSM client creation part into `getSSMValue` where it is actually required does more justice to this method.

Reply:
The client is intentionally kept outside the scope of this method so that it can be reused on subsequent invocations.

Reply:
If you meant subsequent lambda invocations, that is something new to me that we can share local variables across invocations.